docs/2_skeleton.md

```diff
@@ -0,0 +1,169 @@
+   @Version("0.0.1")
+   ```
+
+   To give context about the patch, annotations are used. They serve different but important purposes:
+
+   - Every visible patch **should** be annotated with `@Patch` to be picked up by `PatchBundle` from the [introduction](1_introduction.md). Patches which are not annotated with `@Patch` can be referenced by other patches. We refer to those as _patch dependencies_. Patch dependencies are useful to structure multiple patches.
+
+     Example: _To add settings switches to an app, first, a patch is required which can provide a basic framework for other patches to add their toggles to that app. For that, those patches refer to the dependency patch and use its framework to add their toggles to an app. The [ReVanced patcher](https://github.com/revanced/revanced-patcher) will then first execute the dependency and then the patch itself. The dependency can prepare a preference screen when executed and then initialize the its framework for further use by other patches._
+
+   - Visible patches **should** be annotated with `@Name`. This annotation does not serve any functional purpose. Instead, it allows to refer to the patch with a name. The [ReVanced patches](https://github.com/revanced/revanced-patches) use _Kebab casing_ by convention, but any name can be used for patches. Patches with no `@Patch` annotation do not require the `@Name` annotation, because they are only useable as dependencies for other patches, and therefore are not visible through `PatchBundle`.
+
+   - Visible patches should be annotated with `@Description`. This annotation serves the same purpose as the annotation `@Name`. It is used to give the patch a short description.
+
+   - Patches can be annotated with `@DependsOn`. If the current patch depends on other patches, it can declare them as dependencies.
+
+     Example: _The patch to remove ads needs to patch the bytecode. Additionally it makes use of a second patch, to get rid of resource files in the app which show ads in the app._
+
+   - **All patches** should be annotated with `@Compatibility`. This annotation is the most complex, but **most important** one and serves the purpose of constraining a patch to a package. Every patch is compatible with usually one or more packages. Additionally, the constraint can optionally be extended to versions of the package to discourage the use of the patch with versions outside of the constraint.
+
+     Example: _The patch disables ads for an app. The app regularly updates and the code of the app mutates heavily. In that case the patch might not be compatible for future, untested versions of the app. To discourage the use of the app with other versions than the versions, this patch was confirmed to work on, it is constrained to those versions only._
+
+   - Patches can be annotated with `@Version`.
+
+     > Currently, this annotation does not serve any purpose, but is added to patches by convention, in case a use case has been found.
```

Comment:
Well, there is a use case for it and it's being used for exactly that. It's the fault of the developers that they aren't bumping the versions.

Reply:
A functional purpose is meant. It is currently unused anywhere and does not serve any functional purpose.